CAIPs/caip-222.md

```diff
@@ -0,0 +1,249 @@
+---
+caip: 222
+title: Wallet Authenticate JSON-RPC Method
+author: Pedro Gomes (@pedrouid)
+discussions-to: https://github.com/ChainAgnostic/CAIPs/discussions/222
+status: Draft
+type: Standard
+created: 2022-04-07
+requires: ["2", "10", "74", "122"]
+---
+
+## Simple Summary
+
+This CAIP defines a JSON-RPC method to authenticate wallet ownership/control of a blockchain account.
+
+## Abstract
+
+This proposal provides a simpler user experience by bundling two wallet requests into a single method to provide wallet authentication when exposing blockchain accounts with a cryptographic signature
+
+## Motivation
+
+Prior this standard Wallets were required to establish a JSON-RPC connection with an application (Dapp) to expose blockchain accounts but this required a secondary request to verify ownership of each account to return a cryptographic signature.
+
+There are challenges regarding the flexibility of what can be signed or not without prior knowledge of the blockchain accounts but with the adoption of standards like SIWx ([CAIP-122][caip-122]) we have the ability to define specific parameters that can be signed securely without the accounts being exposed beforehand.
+
+This will accelerate wallet adoption by social media applications which do not require persistent sessions and only need a single cryptographic signature to verify ownership of blockchain accounts.
+
+## Specification
+
+This JSON-RPC method can be requested to a wallet provider without prior knowledge of the blockchain accounts.
+
+The requester will provide parameters required by [CAIP-122][caip-122] plus a CACAO header type as specified by [CAIP-74][caip-74]
+
+The responder will return a signed CACAO with a header type and payload matching the requested parameters
+
+### Request
+
+The application would interface with a provider to make request as follows:
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "wallet_authenticate",
+  "params": {
+      "cacaov": string,
+      "type": string,
+      "chainId": string,
+      "domain": string,
+      "aud": string,
+      "version": string,
+      "nonce": string,
+      "iat": string,
+      "exp": string, // optional
+      "nbf": string, // optional
+      "statement": string, // optional
+      "requestId": string, // optional
+      "resources": string[], // optional
+  }
+}
+```
+
+The JSON-RPC method is labelled as `wallet_authenticate` and expects the following parameters:
+
+- cacaov - Cacao version number
+- type - Cacao header message type
+- chainId - [CAIP-2][]-defined `chainId` to identify specific chain or network.
+- domain - [RFC 4501][rfc 4501] `dnsauthority` that is requesting the signing.
```

Comment:
@awoie we should make sure whatever we land on for SIWE to match here.

@pedrouid the current thinking is that we’ll stay with domain here because that’s what’s already deployed.

In any case, we should make sure the cited RFC matches as well, because IIRC these are different right now.

Reply:
Sounds good! For the record these parameter descriptions were not inherited from SIWE (EIP-4361) but from SIWX (CAIP-122)

CAIP-222 is essentially a combination of CAIP-122 for request and CAIP-74 for response in a JSON-RPC exchange

So there is also going to be some changes inherited from CACAOv3 PR that is currently pending (#179)